avm/res/document-db/mongo-cluster/user/main.bicep

```diff
@@ -0,0 +1,72 @@
+metadata name = 'Azure Cosmos DB for MongoDB (vCore) cluster user'
+metadata description = 'This module creates a user within an Azure Cosmos DB for MongoDB (vCore) cluster. These users are used to connect to the cluster and perform operations using Microsoft Entra authentication.'
+
+@description('Conditional. The name of the parent Azure Cosmos DB for MongoDB (vCore) cluster. Required if the template is used in a standalone deployment.')
+param mongoClusterName string
+
+@description('Optional. Default to current resource group scope location. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. The principal/identity to create as a user on the cluster.')
+param targetIdentity targetPrincipalPropertiesType
+
+@description('Optional. The roles to assign to the user per database. Defaults to the "dbOwner" role on the "admin" database.')
+param targetRoles rolePropertiesType[] = [
+  {
+    database: 'admin'
+    role: 'dbOwner'
+  }
+]
+
+resource mongoCluster 'Microsoft.DocumentDB/mongoClusters@2025-04-01-preview' existing = {
+  name: mongoClusterName
+}
+
+resource user 'Microsoft.DocumentDB/mongoClusters/users@2025-04-01-preview' = {
+  name: targetIdentity.principalId
+  parent: mongoCluster
+  location: location
+  properties: {
+    identityProvider: {
+      type: 'MicrosoftEntraID'
+      properties: {
+        principalType: targetIdentity.?principalType
+      }
+    }
+    roles: [
+      for targetRole in targetRoles: {
+        db: targetRole.database
+        role: targetRole.role
+      }
+    ]
+  }
+}
+
+@export()
+@description('Properties for the identity associated with the user.')
+type targetPrincipalPropertiesType = {
+  @description('Required. The principal (object) ID of the identity to create as a user on the cluster.')
+  principalId: string
+
+  @description('Required. The type of principal to be used for the identity provider.')
+  principalType: 'ServicePrincipal' | 'User'
+}
+
+@export()
+@description('Properties for the role[s] assigned to the user.')
+type rolePropertiesType = {
+  @description('Required. The database to assign the role to.')
+  database: 'admin'
+
+  @description('Required. The role to assign to the user.')
+  role: 'dbOwner'
```

Comment:
Note: This means **only** these 2 values are allowed. This is **not** a default

Reply:
That's a tricky one. It's a union set of allowed values but the PG only allows a single value today.

So, eventually there will be multiple roles. But today, it's a string value where the only allowed value is `admin` for **database** and `dbOwner` for **role**.

Reply:
In the future, I want to be able to set other roles as soon as they're released like:

```bicep
role: 'dbOwner' | 'dbWriter' | 'dbReader'
```

By making it a union field today, my intention was to minimize the breaking changes once they release new roles.

Reply:
Fascination. But all good - just wanted to make sure this was done on purpose. 💪